src/contexts/ServerContext.js

```diff
@@ -145,22 +145,51 @@ export const ServerProvider = ({ children }) => {
                 'Accept': 'application/json',
                 'Authorization': localStorage.getItem('zmt-token')
             }
-        }).then((res) => {
-            let catalog_service_provider = [res.data.zones[0]['icat_server']];
-            let fullServersArray = catalog_service_provider.concat(res.data.zones[0]['servers']);
-            setZoneContext(fullServersArray);
-            setFilteredServers(fullServersArray.slice(0, 10));
-        }).catch((e) => {
-        });
+        })
+    }
+
+    // to query all resources that live on a specific hostname
+    const fetchServerResources = (server_hostname) => {
+        return axios({
+            method: 'GET',
+            url: `${restApiLocation}/query`,
+            headers: {
+                'Authorization': localStorage.getItem('zmt-token')
+            },
+            params: {
+                query_string: `SELECT RESC_NAME WHERE RESC_LOC = '${server_hostname}'`,
+                query_limit: 100,
+                row_offset: 0,
+                query_type: 'general'
+            }
+        })
     }
 
-    const loadCurrServer = (offset, perPage, order, orderBy) => {
+    // load all servers at each render, and iterate through server list to fetch resources which have the same hostname
+    const loadServers = async () => {
+        let zone_report = await loadZoneReport();
+        let catalog_service_provider = [zone_report.data.zones[0]['icat_server']];
+        let fullServersArray = catalog_service_provider.concat(zone_report.data.zones[0]['servers']);
+        for (let curr_server of fullServersArray) {
+            let resource_counts = await fetchServerResources(curr_server['host_system_information']['hostname']);
+            curr_server["resources"] = resource_counts.data.total;
+        }
+        setZoneContext(fullServersArray)
+        setFilteredServers(fullServersArray.slice(0, 10));
+    }
+
+    // handle servers page pagination and sorting
+    const loadCurrServer = async (offset, perPage, order, orderBy) => {
         if (zoneContext !== undefined) {
             let tem_servers = zoneContext;
             let orderSyntax = order === 'asc' ? 1 : -1;
 
             const server_sort_comparator = (a, b) => {
                 switch (orderBy) {
+                    // NOTE: this comparator will ignore 'ip' and all '-' in ip address and concate the number together,
+                    // and edges cases are also handled here.
+                    // e.g. 'ip-172-31-13-194' will be converted to '172031013194'
+                    // 'ip-2-21-13-194' will interpreted as '0020210130194'
```

Comment:
why?   can't we just sort the actual value?   we don't know what someone's hostname could be...

Reply:
Yeah I tried to sort the ip string directly but found out it might not work in some circumstances: "172-3-13-194" will be sorted after "172-21-13-194", as the fifth character '3' is bigger than '2', but it should be sorted before. '172-3-13-194' should be read as '172-003-013-194', so I wrote this helper function. Does that make sense?

Reply:
nopes - don't think about the strings - strings are gonna sort themselves... don't worry about it.

if someone has their real hostname as '172-21-13-194', then ... well, they can find it easy enough.

Reply:
kk. Let me refactor these

Reply:
i see the new code - easy enough.  thanks.